### ISSUE TITLE
Kubernetes Ansible Error: OperationRetry: One or more hosts are unreachable.

### ISSUE THREAD

**Issue by nfisdev** [cloudify-community/blueprint-examples#87]
Hello,

I am trying to use the Kubernetes example blueprint to set up a cluster on OpenStack. 

Cloudify successfully creates all of the nodes, but always fails once it gets to the Ansible script. The exact error I get is "OperationRetry: One or more hosts are unreachable."

I have tried to manually SSH into the two instances  (kube-master, kube-node) but I am unable to
Ive ran it several times and some times  I get connection refused and an Nmap scan shows that there are no open ports on the host, other times I get "permission denied" and an Nmap scan shows that port 22 is open as it should be. 

I am using CentOS7 for the images. I do notice that the instances don't have a key pair in OpenStack, is it supposed to work that way? 

At what point in the process is the SSH keys installed from the deployment inputs? Are they installed through the OpenStack API or through another process?

I am using the OpenStack v3 plugin

Thank you so much for your help, 
Ive been working on this for a while.

This is my input file:
```
username: { get_secret: openstack_username }
password: { get_secret: openstack_password }
auth_url: { get_secret: auth_url }
kubespray_ansible_playbook_path: kubespray/cluster.yml
project_name:  { get_secret: openstack_project_name }
agent_user: centos
region_name: { get_secret: openstack_region }
external_network_id: External
public_subnet_cidr: 10.10.1.0/24
image_id: CentOS7
kubespray_dns_mode: kubedns
client_config_dict: {insecure: true, project_domain_name: { get_secret: project_domain_name }, user_domain_name: { get_secret: user_domain_name }, username: { get_secret: openstack_username }, project_name: { get_secret: openstack_project_name }, password: { get_secret: openstack_password }, auth_url: { get_secret: auth_url }, region_name: { get_secret: openstack_region }}
kubespray_ansible_ssh_common_args: -o ControlMaster=auto -o ControlPersist=30m -o ConnectionAttempts=100 -o UserKnownHostsFile=/dev/null
kubernetes_helm_ansible_playbook_path: ansible-helm/install.yml
kubespray_ansible_playbook_path_scale: kubespray/scale.yml
kubespray_kube_network_plugin: flannel
kubespray_ansible_playbook_path_removal: kubespray/remove-node.yml
flavor_id: mm1.small

```

**Comment by harris419-systologix** [comment]
I am facing the same issue mentioned by @nfisdev. Have been working on this for quite a while, it would be a great relief if someone helps me out of this issue! 

I sincerely appreciate your attention to this matter!

**Comment by natishalom** [comment]
According this
https://github.com/cloudify-cosmo/cloudify-openstack-plugin/issues/329#issuecomment-573239901

You should try using the rpm install and not the QCOW VM

**Comment by harris419-systologix** [comment]
Thank you for your response . 
So you are suggesting that I should not follow this approach:
https://docs.openstack.org/image-guide/centos-image.html 

Instead I should download the iso file to openstack and then create the repo for the centos7 for the rpm package building!

**Comment by natishalom** [comment]
As a workaround yes.
We’ll figure the issue with the image in parallel.

**Comment by harris419-systologix** [comment]
There is another issue in kube-config.yaml:
capabilities:
endpoint:
    value: { concat: [ "https://", { get_attribute: [ kube-master, ip ] }, ":6443" ] }
Capabilities does not support the "concat" feature removing this line gives no error, instead it supports:
capabilities:
    webapp_endpoint:
        description: ip and port of the web application
        value:
            ip: { get_attribute: [webserver_vm, ip] }
            port: { get_property: [webserver, port] }
source: https://docs.cloudify.co/4.5.5/developer/blueprints/spec-capabilities/

**Comment by AdarShaked** [comment]
@harris419-systologix  Can you elaborate which manager version are you using  and which plugins  versions.
I assume you are trying to run the kubespray/openstack.yaml?

**Comment by harris419-systologix** [comment]
> @harris419-systologix Can you elaborate which manager version are you using and which plugins versions.
> I assume you are trying to run the kubespray/openstack.yaml?

@AdarShaked cloudify manager 5.0.0 and the plugins os=3.2.10, ansible=2.8.0, kubernetes=2.7, fabric=1.6.0, utilities=1.22.1
Yes I am trying to execute the kubespray/openstack.yaml

**Comment by AdarShaked** [comment]
Use 5.0.5 manager.
